jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -927,16 +957,150 @@ static String formatExpires(Instant expires)
      *
-     * @param expires an instant in the RFC 1123 string format
+     * @param date an instant in the string format supported by RFC6265 HTTP State Management
      * @return an {@link Instant} parsed from the given string
      */
-    static Instant parseExpires(String expires)
+    static Instant parseCookieDate(String date)
+    {
+        Objects.requireNonNull(date, "Expires string cannot be null");
+
+        int year = -1;
+        int month = -1;
+        int day = -1;
+        int hour = -1;
+        int minute = -1;
+        int second = -1;
+
+        try
+        {
+            int tokenCount = 0;
+            Iterator<String> tokenIter = Immutable.DATE_TOKENIZER.tokenize(date);
+            while (tokenIter.hasNext())
+            {
+                String token = tokenIter.next();
+                // ensure we don't exceed the number of expected tokens.
+                if (++tokenCount > 6)
+                {
+                    // This is a horribly bad syntax / format
+                    throw new IllegalStateException("Too many delimiters for a Date format");
+                }
+
+                if (token.isBlank())
+                    continue; // skip blank tokens
+
+                // RFC 6265 - Section 5.1.1 - Step 2.1 - time (00:00:00)
+                // if (hour == (-1) && Immutable.PATTERN_TIME.matcher(token).matches())
+                if (hour == (-1) && token.length() == 8 && token.charAt(2) == ':' && token.charAt(5) == ':')
+                {
+                    hour = StringUtil.toInt(token, 0);
+                    minute = StringUtil.toInt(token, 3);
+                    second = StringUtil.toInt(token, 6);
+                    continue;
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.2
+                if (day == (-1) && token.length() <= 2)
+                {
+                    day = StringUtil.toInt(token, 0);
+                    continue;
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.3
+                if (month == (-1) && token.length() == 3)
+                {
+                    Integer m = Immutable.MONTH_CACHE.getBest(token);
+                    if (m != null)
+                    {
+                        month = m;
+                        continue;
+                    }
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.4
+                if (year == (-1))
+                {
+                    if (token.length() <= 2)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    else if (token.length() == 4)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    continue;
+                }
+            }
+        }
+        catch (Throwable t)
+        {
+            throw new DateTimeSyntaxException("Unable to parse date: " + date, t);
```

Comment:
I still think technically this is just an ignore and continue.... but fail early is probably better.
Does it still pass tests with an ignore and continue?

Reply:
done